src/base/tests/test_util.py

```diff
@@ -859,14 +883,16 @@ def test_invert_boolean_field(self):
             util.rename_field(cr, model, new_name, old_name)
 
         util.invalidate(fltr)
-        self.assertEqual(literal_eval(fltr.domain), ["!", (old_name, "=", True)])
+        expected = [(old_name, "!=", True)] if USE_ORM_DOMAIN else ["!", (old_name, "=", True)]
+        self.assertEqual(literal_eval(fltr.domain), expected)
 
         # invert with same name; will invert domains and data
         with mock.patch.object(cr, "commit", lambda: ...):
             util.invert_boolean_field(cr, model, old_name, old_name)
 
         util.invalidate(fltr)
-        self.assertEqual(literal_eval(fltr.domain), ["!", "!", (old_name, "=", True)])
+        expected = ["!", (old_name, "!=", True)] if USE_ORM_DOMAIN else ["!", "!", (old_name, "=", True)]
```

Comment:
Note that it show that the domain simplification could still be improved.

Reply:
Actually, using the Domain results in `[(old_name, '=', True)]`, but the inverse boolean just does `[..., *(~Domain(some_leaf)), ...]` and does not normalize the whole domain.